CodePush.js

```diff
@@ -170,8 +170,20 @@ function setUpTestDependencies(testSdk, providedTestConfig, testNativeBridge) {
   if (testNativeBridge) NativeCodePush = testNativeBridge;
 }
 
+// This function allows calls to syncInternal to be chained on to each other so that they do not
+// interleave in the event that sync() is called multiple times.
+const sync = (() => {
```

Comment:
I definitely don't see any point in allowing parallel requests. I also think rejecting the sync is better. The metaphor I'm thinking of is when an email client is syncing with the email server, when you do a pull-to-refresh in the app for example, it would block you from doing the pull-to-refresh again until the existing one is completed.